cmake/DynamicVersion.cmake

```diff
@@ -0,0 +1,661 @@
+# Retrieved from https://github.com/LecrisUT/CMakeExtraUtils/blob/v0.4.0/cmake/DynamicVersion.cmake
```

Comment:
Ideally, this utility function should not be maintained in this project. Is there any plan to migrate it?

Reply:
Yes, and this one particularly is kind of tricky because it needs to be called before `project`. The plan for this module is:
- Package it to Fedora+EPEL. Would need help for Debian distros
- Wait until we can bump CMake minimum to 3.24 (have to wait for RHEL10 most likely :( ) so that we can use `FetchContent(FIND_PACKAGE_ARGS)`, then I will import CMakeExtraUtils directly (otherwise I would have to bundle [this compatibility module](https://github.com/LecrisUT/CMakeExtraUtils/blob/compat/FetchContent/cmake/compat/CEU_FetchContent.cmake))
- Migrate these functions to CMake upstream as well, and use the former for compatibility until that CMake version makes it to LTS distros

Reply:
I see. I am glad to hear that there is a path to FetchContent.